### PR TITLE
[sonic-app-ext] support app extensions installation during build

### PR DESCRIPTION
**Pull request by stepanblyschak**
Signed-off-by: Stepan Blyschak <stepanb@mellanox.com>

#### Why I did it

To support building DHCP relay as extension and installing it during build time.

#### How I did it

Created infrastructure. Users need to define their packages in rules/sonic-packages.mk

#### How to verify it

Together with https://github.com/Azure/sonic-buildimage/pull/6531

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Review by tahmed-dev — DISMISSED** [review]
LGTM. just a minor comment

**Comment by liat-grozovik** [comment]
@stepanblyschak please handle conflicts and rerun tests.

**Comment by stepanblyschak** [comment]
There is some problem with build I doubt it is related to this change, VS build fails:
```
dpkg-checkbuilddeps: error: Unmet build dependencies: python3-pytest
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)
Makefile:11: recipe for target '/sonic/target/debs/stretch/frr_7.5.1-sonic-0_amd64.deb' failed

```

The same frr package has been built succsfully for mellanox and broadcom. Also, kvmtests just stuck and do not retrigger.

**Comment by liat-grozovik** [comment]
@stepanblyschak can you please handle the new conflict?
@tahmed-dev  and @renukamanavalan could you please help to review?

File: files/scripts/swss.sh [hunk 1]
```diff
@@ -3,6 +3,15 @@
 DEPENDENT="radv dhcp_relay"
 MULTI_INST_DEPENDENT="teamd"
 
+# Update dependent list based on other packages requirements
```
Comment (renukamanavalan) [inline files/scripts/swss.sh:6]: Why only swss.sh?

Comment (stepanblyschak) [reply files/scripts/swss.sh:6]: Needed by dhcp-relay. Could be in any other script as well on demand.

Comment (stepanblyschak) [reply files/scripts/swss.sh:6]: Actually, the right way is to have a dependency script autogenerated, for others (bgp, teamd, frr) it means once they become packages the will automatically have it.

**Comment by liat-grozovik** [comment]
@renukamanavalan and @tahmed-dev can you please approve?

**Comment by renukamanavalan** [comment]
@tahmed-dev can you please approve?
I am good with this PR.

File: files/scripts/swss.sh [hunk 2]
```diff
@@ -1,8 +1,17 @@
 #!/bin/bash
 
-DEPENDENT="radv dhcp_relay"
+DEPENDENT="radv"
```
[the remark this answers is not in the document]
Comment (tahmed-dev) [reply files/scripts/swss.sh]: @stepanblyschak , can you please resolve this nit comment?

[the remark this answers is not in the document]
Comment (stepanblyschak) [reply files/scripts/swss.sh]: It is already resolved

**Review by tahmed-dev — APPROVED** [review]

File: slave.mk [hunk 3]
```diff
@@ -748,6 +750,7 @@ $(addprefix $(TARGET_PATH)/, $(DOCKER_IMAGES)) : $(TARGET_PATH)/%.gz : .platform
 		$(eval export $(subst -,_,$(notdir $($*.gz_PATH)))_whls=$(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_PYTHON_WHEELS)))\n" | awk '!a[$$0]++'))
 		$(eval export $(subst -,_,$(notdir $($*.gz_PATH)))_dbgs=$(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_DBG_PACKAGES)))\n" | awk '!a[$$0]++'))
 		$(eval export $(subst -,_,$(notdir $($*.gz_PATH)))_pkgs=$(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_APT_PACKAGES)))\n" | awk '!a[$$0]++'))
+		if [ -d $($*.gz_PATH)/cli-plugin-tests/ ]; then pushd $($*.gz_PATH)/cli-plugin-tests; pytest-$($(SONIC_UTILITIES_PY3)_PYTHON_VERSION) -v $(LOG); popd; fi
```
Comment (qiluo-msft) [inline slave.mk:753]: >pytest [](http://example.com/codeflow?start=88&length=6)

Could you move python unit test out of the process of building a docker image target? We should treat this package as a normal standalone python wheel target.

**Pull request merged**